source/gui/settingsDialogs.py

```diff
@@ -2527,6 +2527,13 @@ def __init__(self, parent):
 		self.keyboardSupportInLegacyCheckBox.defaultValue = self._getDefaultValue(["terminals", "keyboardSupportInLegacy"])
 		self.keyboardSupportInLegacyCheckBox.Enable(winVersion.isWin10(1607))
 
+		# Translators: This is the label for a checkbox in the
+		#  Advanced settings panel.
+		label = _("Detect changes by c&haracter when supported")
+		self.useDMPWhenSupportedCheckBox = terminalsGroup.addItem(wx.CheckBox(self, label=label))
+		self.useDMPWhenSupportedCheckBox.SetValue(config.conf["terminals"]["useDMPWhenSupported"])
+		self.useDMPWhenSupportedCheckBox.defaultValue = self._getDefaultValue(["terminals", "useDMPWhenSupported"])
+

```

Comment:
It would be better to make this a 3 way selection: "default (difflib)", "difflib", "dmp"

This allows us to change the default without affecting a users explicit preferences.

Reply:
I don't expect the default to change here (this isn't a feature flag, but rather an option to allow users to disable DMP should it break something). I expect that this option will remain long term.

Reply:
I'd still prefer this to be three way. You may not expect to change the default, but that is exactly what I'd like to account for: the unexpected. With a two way option, changing the default guarantees breaking the config. A default allow us to give a recommendation, and later change that recommendation. This approach is quite experimental, I wouldn't expect to get it right the first time. A third implementation might even be tried (EG python extension using GPL2 xdiff from git)

While here, please set the default to be difflib until this change has proven itself (at least on alpha)